### ISSUE TITLE
Potential corruption when using ccache & xlc

### ISSUE THREAD

**Issue by dnakamura** [eclipse/omr#4594]
tl;dr:
If using ccache with xlc, changes to files which do not contain any code after preprocessing (eg `omrcfg.h`) are not properly detected, leading to false cache hits and a potentially corrupted binary

The problem occurs because of the way that ccache determines which files are included (and thus should be added to the hash identifying a compilation result. Te determine which files are included ccache runs the source through the pre-processor, and scrapes the result for `#line` directives. However, it appears that xlc is stripping excess whitespace from the pre-processed output. This means that if, after pre-processing a header file only contains whitespace, it does not get a `#line` directive in the the output, and the file is ignored when ccache is checking for a cache hit. The most obvious case where this could be an issue is with `omrcfg.h` (and indeed, appears to be responsible for #4539)

**Comment by fjeremic** [comment]
Yikes. Is there any obvious solution to this other than just disabling ccache with xlc?

**Comment by rwy0717** [comment]
AIX is one of our slower builds, and ccache saves us a lot of time/resources. Since this particular error is infrequent, I think we should leave ccache in the builds. If a developer sees funny AIX-only failures, ping me personally and I will blow away the cache. I think we will have a real fix soon. @dnakamura suggested that newer versions of ccache have alternative methods to track file dependencies.

**Comment by rwy0717** [comment]
In #4632, we fixed this for `omrcfg.h`, which is the only case where this issue actually occurred. Closing as "solved", but we have to be aware that this could happen for a different header down the road.
